src/boot/efi/stub.c

```diff
@@ -440,46 +442,52 @@ static EFI_STATUS load_addons(
+                        struct PeSectionDescriptor *section_cmdline = pe_bsearch_section(sections, n_sections, unified_sections[UNIFIED_SECTION_CMDLINE]);
+                        if (section_cmdline) {
+                                _cleanup_free_ char16_t *tmp = TAKE_PTR(cmdline),
+                                                        *extra16 = xstrn8_to_16((char *)loaded_addon->ImageBase + section_cmdline->offset,
+                                                                                section_cmdline->size);
+                                cmdline = xasprintf("%ls%ls%ls", strempty(tmp), isempty(tmp) ? u"" : u" ", extra16);
+                        }
                 }
 
-                if (ret_dt_bases && szs[UNIFIED_SECTION_DTB] > 0) {
-                        dt_sizes = xrealloc(dt_sizes,
-                                            n_dt * sizeof(size_t),
-                                            (n_dt + 1)  * sizeof(size_t));
-                        dt_sizes[n_dt] = szs[UNIFIED_SECTION_DTB];
-
-                        dt_bases = xrealloc(dt_bases,
-                                            n_dt * sizeof(void *),
-                                            (n_dt + 1) * sizeof(void *));
-                        dt_bases[n_dt] = xmemdup((uint8_t*)loaded_addon->ImageBase + addrs[UNIFIED_SECTION_DTB],
-                                                 dt_sizes[n_dt]);
-
-                        dt_filenames = xrealloc(dt_filenames,
-                                                n_dt * sizeof(char16_t *),
-                                                (n_dt + 1) * sizeof(char16_t *));
-                        dt_filenames[n_dt] = xstrdup16(items[i]);
-
-                        ++n_dt;
+                if (ret_dt_bases) {
+                        struct PeSectionDescriptor *section_dtb = pe_bsearch_section(sections, n_sections, unified_sections[UNIFIED_SECTION_DTB]);
+                        if (section_dtb) {
+                                dt_sizes = xrealloc(dt_sizes,
+                                                    n_dt * sizeof(size_t),
+                                                    (n_dt + 1)  * sizeof(size_t));
+                                dt_sizes[n_dt] = section_dtb->size;
+
+                                dt_bases = xrealloc(dt_bases,
+                                                    n_dt * sizeof(void *),
+                                                    (n_dt + 1) * sizeof(void *));
+                                dt_bases[n_dt] = xmemdup((uint8_t*)loaded_addon->ImageBase + section_dtb->offset,
+                                                         dt_sizes[n_dt]);
+
+                                dt_filenames = xrealloc(dt_filenames,
+                                                        n_dt * sizeof(char16_t *),
+                                                        (n_dt + 1) * sizeof(char16_t *));
+                                dt_filenames[n_dt] = xstrdup16(items[i]);
```

Comment:
we should look into maybe reused PeSectionDescriptor eventually to keep track of the dt sections too, i.e. add a new "filename" field to PeSectionDescriptor, and then just keep one array of PeFileDescriptor for the dt stuff here, that tracks the PE sections of all addons with references back to them if we need them.

but that#s for a later time